### ISSUE TITLE
Talkback is currently announcing incorrect information. When a list item is selected or deselected, it announces the details of the previous item instead of the current one: A11y_.NET 9 template_Projects screen_Talkback.

### ISSUE THREAD

**Issue by keerthiduvvuri** [dotnet/maui#30833]
### GitHub Tags:
#A11yTCS; #A11yMAS; #WCAG1.3.1; #A11ySev2; #Android; #BM_.NET maui_Android_JULY2025; #.NET MAUI SDKs; #Device; #Info and Relationship; #AINotInscope; #Talkback;

### Environment Details:
Application Name: developer balance version .NET 10
OS: Android
Model Name: Google Pixel 8
Android version: 15

### Repro Steps:

1. Install and open "Developer balance" app.
2. "Home Screen" will open.
3. Swipe till project section, and Double tap on any project, Project screen will open, 
4. Swipe till Categories and Choose a category from the list of category.
5. Swipe till Tags and Choose a tag from the list of tags.
6. We can select the task from the existing task list.
7. Swipe till new task button in the project screen.
8. Observe that Talkback is currently announcing incorrect information. When a list item is selected or deselected, it announces the details of the previous item instead of the current one.

### Actual Result:
Talkback is currently announcing incorrect information. When a list item is selected or deselected, it announces the details of the previous item instead of the current one.

### Expected Result:
Talkback should be announcing correct information. When a list item is selected or deselected, it should not be announces the details of the previous item instead of the current one.

### User Impact:
Screen reader users will find it difficulty of knowing or accessing the complete information of those list items if, Talkback is currently announcing incorrect information. When a list item is selected or deselected, it announces the details of the previous item instead of the current one.

### Attachment:

https://github.com/user-attachments/assets/17509ea9-c172-4681-bdd8-d70b08b07e16

**Comment by ghost** [comment]
This issue has been verified in Visual Studio Code 1.102.1 with MAUI versions 9.0.0 & 9.0.90. It can be reproduced on the Android platform

**Comment by PoojaNamde** [comment]
#Closed; #A11yFixed; #Regressed:09-19-25;

Verified the issue in the latest build. Issue is fixed hence closing the bug.
